testcrate/tests/conv.rs

```diff
@@ -130,3 +140,118 @@ fn float_to_int() {
         );
     });
 }
+
+macro_rules! conv {
+    ($fX:ident, $fD:ident, $fn:ident, $apfloatX:ident, $apfloatD:ident) => {
+        fuzz_float(N, |x: $fX| {
+            let tmp0: $apfloatD = $apfloatX::from_bits(x.to_bits().into())
+                .convert(&mut false)
+                .value;
+            let tmp0 = $fD::from_bits(tmp0.to_bits().try_into().unwrap());
+            let tmp1: $fD = $fn(x);
+            if !Float::eq_repr(tmp0, tmp1) {
+                panic!(
+                    "{}({x:?}): apfloat: {tmp0:?}, builtins: {tmp1:?}",
+                    stringify!($fn)
+                );
+            }
+        });
+    };
+}
+
+macro_rules! extend {
+    ($fX:ident, $fD:ident, $fn:ident) => {
+        fuzz_float(N, |x: $fX| {
+            let tmp0 = x as $fD;
+            let tmp1: $fD = $fn(x);
+            if !Float::eq_repr(tmp0, tmp1) {
+                panic!(
+                    "{}({}): std: {}, builtins: {}",
+                    stringify!($fn),
+                    x,
+                    tmp0,
+                    tmp1
+                );
+            }
+        });
+    };
+}
+
+// PowerPC tests are failing on LLVM 13: https://github.com/rust-lang/rust/issues/88520
+#[cfg(not(target_arch = "powerpc64"))]
+#[test]
+fn float_extend() {
+    use compiler_builtins::float::extend::__extendsfdf2;
+
+    extend!(f32, f64, __extendsfdf2);
+    conv!(f32, f64, __extendsfdf2, Single, Double);
+    #[cfg(not(feature = "no-f16-f128"))]
+    {
+        use compiler_builtins::float::extend::{
+            __extenddftf2, __extendhfsf2, __extendhftf2, __extendsftf2, __gnu_h2f_ieee,
+        };
+        // FIXME: Also do extend!() for `f16` and `f128` when builtins are in nightly
```

Comment:
Minor nit, could you change these couple fixmes to `FIXME(f16_f128)`? To be consistent with fixmes related to these features in the compiler.